cirq-core/cirq/circuits/circuit_operation.py

```diff
@@ -256,7 +256,7 @@ def mapped_circuit(self, deep: bool = False) -> 'cirq.Circuit':
             circuit = protocols.with_measurement_key_mapping(circuit, self.measurement_key_map)
         if self.param_resolver:
             circuit = protocols.resolve_parameters(circuit, self.param_resolver, recursive=False)
-        if self.repetition_ids:
+        if self.repetition_ids is not None:
```

Comment:
It seems a bit odd to me to couple the behavior at 0 repetitions to `self.repetition_ids`, especially since if `self.use_repetition_ids` is `False` (which may become the default in the future), then having `self.repetition_ids == None` would be the natural thing for _any_ number of repetitions. It might be cleaner to have an early special case (which can happen before we do anything else in this function since all the other mapping logic becomes irrelevant with zero reps):
```python
def mapped_circuit(self, ...):
    if self.repetitions == 0:
        return self.circuit * 0
    circuit = self.circuit.unfreeze()
    ...
```

Reply:
sg, I added the early return code. I left the `is not None` check for consistency with everywhere else in the file we check `self.repetition_ids`, and also left the change in `default_repetition_ids` since there's some logic in the `repeat` function that also treats `None` as meaning exactly one repetition.